main.tex

```diff
@@ -2,7 +2,7 @@ \chapter{Introduction}
 
 \section{Background}
 
-It is generally useful for software to be able to automatically determine the
+It is generally useful for software to be able to programmatically determine the
```

Comment:
Tim, it was "automatically " in this sentence, is "programmatically"  closer to the given mechanism of providing System Descriptions?